### ISSUE TITLE
The feature become Nan after the “conv_input” layer in the "VoxelResBackBone8x"

### ISSUE THREAD

**Issue by twn29004** [open-mmlab/OpenPCDet#928]
When I was training centernet with my own loss function, I found that in VoxelResBack8x, the features became nan after going through the conv_input layer.

https://github.com/open-mmlab/OpenPCDet/blob/846cf3ed21c274f95f041e7e21fd039ca5c99a1f/pcdet/models/backbones_3d/spconv_backbone.py#L252-L260

And the conv_input define is:
https://github.com/open-mmlab/OpenPCDet/blob/846cf3ed21c274f95f041e7e21fd039ca5c99a1f/pcdet/models/backbones_3d/spconv_backbone.py#L191-L195

And I have checked the input voxel_feature is not Nan.  Can anyone give me a suggestion？I don't know what to do with it.  Below are some results from my debugging process.
![](https://blog-image-twn29004.oss-cn-chengdu.aliyuncs.com/blog-img/202204141957411.png)
![](https://blog-image-twn29004.oss-cn-chengdu.aliyuncs.com/blog-img/202204141957618.png)

**Comment by twn29004** [comment]
![](https://blog-image-twn29004.oss-cn-chengdu.aliyuncs.com/blog-img/202204142015669.png)

**Comment by twn29004** [comment]
My pcdet version is 0.5.1+eed89a4, and the spconv version is spconv-cu114  2.1.21

**Comment by twn29004** [comment]
I have found the problem. is the problem with my loss function.

**Comment by JianyyuWang** [comment]
Do you solve this problem？

**Comment by twn29004** [comment]
> 
There are some error in my loss function. May be you can add the following code in the training function to check the probelm.
```python
import torch  
# forward 
torch.autograd.set_detect_anomaly(True)  
# TODO
# backward  
with torch.autograd.detect_anomaly():  
    loss.backward()  
```

**Comment by study1994** [comment]
![image](https://github.com/open-mmlab/OpenPCDet/assets/25917053/f9adabd6-c5cc-4d0e-a17f-afbf099c486e)
Same problem, following the workaround should not be a loss problem
